app/models/provider.rb

```diff
@@ -1,4 +1,8 @@
 class Provider < Base
   belongs_to :recruitment_cycle, param: :recruitment_cycle_year
   has_many :courses, param: :course_code
+
+  def full_address
+    [address1, address2, address3, address4, postcode].select(&:present?).join("<br> ").html_safe
+  end
```

Comment:
Feels like this should be in the decorator, since it deals with outputting HTML.

Reply:
Same as frontend, leaving as is

Reply:
The C# frontend? Why are we replicating it's application architecture? I didn't even realise it had decorators.

Reply:
`manage-courses-frontend`. Remember all this code is copy and paste

Reply:
Ah, I see. It's wrong there then too.

Reply:
Since decorators wrap existing objects, fixing this is as simple as moving this to the decorator. Unless we're not instantiating decorator objects for views correctly, in which case that's a bigger problem that should fixed now before we have a bunch of view code shoved into the model.